R/hiper_glm.R

```diff
@@ -1,31 +1,104 @@
+
+####################### Helper functions ######################
+
+########## For cholesky ########
+## Find linear regression coefficients estimates via MLE,
+## with cholesky decomposition while solving matrices
+chol_finder <- function(design, outcome){
+  ## With least-square, we have,
+  ## beta_est = (X^T X) X^T y.
+  ## Which is solving beta_est in (X^T X) beta_est = X^T y.
+  ## This is equivalent to solving Ax = b,
+  ## where, A and b are the following
+
+  A <- t(design) %*% design
+  b <- t(design) %*% outcome
+
+  ## solve "Ax = b" question via cholesky decomp
+  L <- chol(A)
+  beta_est <- backsolve(t(L), forwardsolve(L, t(A) %*% b))
+  return(c(beta_est))
+}
+
+
+############ FOR BFGS ############
+## The log-likelihood function
+gaussian_logLi <- function(beta, X, Y, noise_var = 1){
```

Comment:
This is a bit of a awkward function name,  mixing in the `L` in an otherwise snake case scheme. would just do `gaussian_loglik` or something